DAHDI hardware with hardware DTMF detection would get accidentally muted
after a DTMF is sent.

Upstream issue: https://issues.asterisk.org/view.php?id=14761

--- a/channels/chan_dahdi.c
+++ b/channels/chan_dahdi.c
@@ -4710,6 +4710,7 @@ static struct ast_frame *dahdi_handle_ev
 			/* absorb event */
 		} else {
 #endif
+			dahdi_confmute(p, 0);
 			p->subs[idx].f.frametype = AST_FRAME_DTMF_END;
 			p->subs[idx].f.subclass = res & 0xff;
 #ifdef HAVE_PRI
